hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/CallFeature.java

```diff
@@ -84,23 +119,106 @@ public static String[] splitAddresses(String result) {
+        estimateContractAddress = deployedEstimatePrecompileContract.contractId().toSolidityAddress();
+        admin = tokenClient.getSdkClient().getExpandedOperatorAccountId();
         receiverAccountId = accountClient.getAccount(AccountNameEnum.BOB);
+        receiverAccountAlias = receiverAccountId.getPublicKey().toEvmAddress().toString();
+        secondReceiverAccount = accountClient.getAccount(AccountNameEnum.DAVE);
+        fungibleTokenId = tokenClient.getToken(FUNGIBLE).tokenId();
+        fungibleKycUnfrozenTokenId = tokenClient.getToken(FUNGIBLE_KYC_UNFROZEN).tokenId();
+        nonFungibleTokenId = tokenClient.getToken(NFT).tokenId();
+    }
+
+    @Given("I mint a NFT")
+    public void mintNft() {
+        networkTransactionResponse = tokenClient.mint(nonFungibleTokenId, RandomUtils.nextBytes(4));
+    }
+
+    @And("I approve and transfer FUNGIBLE token to receiver account")
+    public void approveAndTransferFungibleTokenToReceiver() {
+        accountClient.approveToken(fungibleTokenId, receiverAccountId.getAccountId(), 1L);
+        networkTransactionResponse = tokenClient.transferFungibleToken(
+                fungibleTokenId,
+                tokenClient.getSdkClient().getExpandedOperatorAccountId(),
+                receiverAccountId.getAccountId(),
+                receiverAccountId.getPrivateKey(),
+                1L);
+    }
+
+    @And("I approve and transfer FUNGIBLE token to the precompile contract")
+    public void approveAndTransferFungibleTokenToPrecompile() {
+        networkTransactionResponse = tokenClient.transferFungibleToken(
+                fungibleTokenId,
+                admin,
+                AccountId.fromString(deployedPrecompileContract.contractId().toString()),
+                receiverAccountId.getPrivateKey(),
+                10L);
+    }
+
+    @And("I approve and transfer NFT token to receiver account")
+    public void approveAndTransferNftTokenToReceiver() {
+        var ntfId = new NftId(nonFungibleTokenId, 1L);
+        accountClient.approveNft(ntfId, receiverAccountId.getAccountId());
+        networkTransactionResponse = tokenClient.transferNonFungibleToken(
+                nonFungibleTokenId,
+                tokenClient.getSdkClient().getExpandedOperatorAccountId(),
+                receiverAccountId.getAccountId(),
+                List.of(1L),
+                receiverAccountId.getPrivateKey());
+    }
+
+    @And("I associate FUNGIBLE token to receiver account")
+    public void associateFungibleTokenToReceiver() {
+        var fungibleTokenStatus = mirrorClient.getTokenRelationships(receiverAccountId.getAccountId(), fungibleTokenId).getTokens().size();
+        var nftStatus = mirrorClient.getTokenRelationships(receiverAccountId.getAccountId(), nonFungibleTokenId).getTokens().size();
+
+        //if the tokens are already associated from the parallel tests run skip the association
+        if(fungibleTokenStatus == 0){
```

Comment:
nit: Should adjust your IDE save actions to auto format. Also, ensure the git hook is running to format files via spotless. Sometimes git UI tools don't respect the git hook.

```suggestion
        if (fungibleTokenStatus == 0) {
```